[DOCS] Add data streams to API conventions (#58695)
elastic/elasticsearch: docs/reference/api-conventions.asciidoc

* <<url-access-control>>

[[multi-index]]
=== Multi-target syntax

Most APIs that accept a `<data-stream>`, `<index>`, or `<target>` request path
parameter also support _multi-target syntax_.

> In multi-target syntax, you can use a comma-separated list (e.g.,
> `test1,test2,test3`) to execute a request across multiple resources, such as
> data streams, indices, or index aliases. You can also use
> https://en.wikipedia.org/wiki/Glob_(programming)[glob-like] wildcard (`*`)
> expressions (e.g., `test*` or `*test` or `te*t` or `*test*`) to target any
> resources that match the pattern.
> 
> You can exclude targets using the `-` character (e.g., `test*,-test3`).

Reviewer: I'd probably take the opportunity to get rid of the e.g.s. 

```suggestion
In multi-target syntax, you can use a comma-separated list to execute a request across multiple resources, such as
data streams, indices, or index aliases: `test1,test2,test3`. You can also use
https://en.wikipedia.org/wiki/Glob_(programming)[glob-like] wildcard (`*`)
expressions to target any
resources that match the pattern: `test*` or `*test` or `te*t` or `*test*.

You can exclude targets using the `-` character: `test*,-test3`.
```

The defaults settings for the above parameters depend on the API being used.

Some multi-target APIs that can target indices also support the following query
string parameter:

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=ignore_throttled]

> NOTE: Single index APIs, such as the <<docs>> and the

Reviewer: ```suggestion
NOTE: Single index APIs, such as the <<docs>> and
```

The defaults settings for the above parameters depend on the API being used.

Some multi-target APIs that can target indices also support the following query
string parameter:

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=ignore_throttled]

NOTE: Single index APIs, such as the <<docs>> and the
<<indices-aliases,single-index `alias` APIs>>, do not support multi-target
syntax.

[[date-math-index-names]]
=== Date math support in target names

Most APIs that accept a `<data-stream>`, `<index>`, or `<target>` request path

> parameter support _date math_.

Reviewer 2: Just a heads-up, but data stream names don't support date math since a single data stream is supposed to handle a potentially infinite stream of data and the data stream's backing indices are named according to their generation and not the date on which they were created.

Author: Thanks @danhermann. I've removed my changes to this section/page.

Technically, you can still search a data stream name using date math. For example, `GET /%3Clogs-%7Bnow%2Fd%7D%3E/_search` would search the `logs-2020.06.30` data stream.

But first users would need to create a data stream with a date component , which we probably shouldn't encourage. It seems like a corner case we'd be better off not mentioning.

=== URL-based access control

Many users use a proxy with URL-based access control to secure access to
{es} data streams and indices. For <<search-multi-search,multi-search>>,
<<docs-multi-get,multi-get>>, and <<docs-bulk,bulk>> requests, the user has
the choice of specifying a data stream or  index in the URL and on each individual request
within the request body. This can make URL-based access control challenging.

> To prevent the user from overriding the data stream or index which has been specified in the
> URL, add this setting to the `elasticsearch.yml` file:
> 
>     rest.action.multi.allow_explicit_index: false

Reviewer: ```suggestion
To prevent the user from overriding the data stream or index specified in the
URL, set `rest.action.multi.allow_explicit_index` to `false` in `elasticsearch.yml`.

```

> The default value is `true`, but when set to `false`, {es} will
> reject requests that have an explicit data stream or index specified in the request body.

Reviewer: ```suggestion
This causes  {es} to
reject requests that explicitly specfiy a data stream or index in the request body.
```
